### ISSUE TITLE
Can't export to Windows from Raspbian

### ISSUE THREAD

**Issue by liquid600pgm** [processing/processing#5287]
## Description

If you try to export a sketch from Raspbian to Windows it will fail.

## Expected Behavior

It should normally export a Windows .exe, but instead shows an error message in the console (see below).

## Current Behavior

On Windows it will just export. On Raspbian, this is the log:

```
Compiling resources
/home/pi/Desktop/processing-3.3.5/modes/java/application/launch4j/bin/windres: 1: /home/pi/Desktop/processing-3.3.5/modes/java/application/launch4j/bin/windres: Syntax error: word unexpected (expecting ")")
Generated resource file...

LANGUAGE 0, 1
2 RCDATA BEGIN "1.8.0_74\0" END
18 RCDATA BEGIN "1\0" END
30 RCDATA BEGIN "2\0" END
12 RCDATA BEGIN "-Djna.nosys=true -Djava.library.path=""%EXEDIR%\\lib""\0" END
1 ICON DISCARDABLE "/home/pi/Desktop/processing-3.3.5/modes/java/application/sketch.ico"
21 RCDATA BEGIN "http://java.com/download\0" END
20 RCDATA BEGIN "32\0" END
101 RCDATA BEGIN "An error occurred while starting the application.\0" END
102 RCDATA BEGIN "This application was configured to use a bundled Java Runtime Environment but the runtime is missing or corrupted.\0" END
103 RCDATA BEGIN "This application requires a Java Runtime Environment\0" END
104 RCDATA BEGIN "The registry refers to a nonexistent Java Runtime Environment installation or the runtime is corrupted.\0" END
15 RCDATA BEGIN "exCircuit\0" END
16 RCDATA BEGIN "lib\\exCircuit.jar;lib\\core.jar;lib\\jogl-all.jar;lib\\gluegen-rt.jar;lib\\jogl-all-natives-windows-i586.jar;lib\\gluegen-rt-natives-windows-i586.jar\0" END


windows:


BUILD FAILED
/home/pi/sketchbook/exCircuit/application.windows32/launch4j-build.xml:5: net.sf.launch4j.BuilderException: net.sf.launch4j.ExecException: Exec failed (2): /home/pi/Desktop/processing-3.3.5/modes/java/application/launch4j/bin/windres --preprocessor=cat -J rc -O coff -F pe-i386 /tmp/launch4j2560377606394614871rc /tmp/launch4j6100233142982000968o
	at net.sf.launch4j.Builder.build(Builder.java:149)
	at net.sf.launch4j.ant.Launch4jTask.execute(Launch4jTask.java:111)
	at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:291)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:106)
	at org.apache.tools.ant.Task.perform(Task.java:348)
	at org.apache.tools.ant.Target.execute(Target.java:390)
	at org.apache.tools.ant.Target.performTasks(Target.java:411)
	at org.apache.tools.ant.Project.executeSortedTargets(Project.java:1399)
	at org.apache.tools.ant.Project.executeTarget(Project.java:1368)
	at processing.mode.java.JavaBuild.buildWindowsLauncher(JavaBuild.java:1222)
	at processing.mode.java.JavaBuild.exportApplication(JavaBuild.java:1093)
	at processing.mode.java.JavaBuild.exportApplication(JavaBuild.java:704)
	at processing.mode.java.JavaMode.handleExportApplication(JavaMode.java:239)
	at processing.mode.java.JavaEditor.exportApplicationPrompt(JavaEditor.java:944)
	at processing.mode.java.JavaEditor.handleExportApplication(JavaEditor.java:574)
	at processing.mode.java.JavaEditor$3.actionPerformed(JavaEditor.java:197)
	at javax.swing.AbstractButton.fireActionPerformed(AbstractButton.java:2022)
	at javax.swing.AbstractButton$Handler.actionPerformed(AbstractButton.java:2348)
	at javax.swing.DefaultButtonModel.fireActionPerformed(DefaultButtonModel.java:402)
	at javax.swing.DefaultButtonModel.setPressed(DefaultButtonModel.java:259)
	at javax.swing.AbstractButton.doClick(AbstractButton.java:376)
	at javax.swing.plaf.basic.BasicMenuItemUI.doClick(BasicMenuItemUI.java:833)
	at javax.swing.plaf.basic.BasicMenuItemUI$Handler.mouseReleased(BasicMenuItemUI.java:877)
	at java.awt.Component.processMouseEvent(Component.java:6533)
	at javax.swing.JComponent.processMouseEvent(JComponent.java:3324)
	at java.awt.Component.processEvent(Component.java:6298)
	at java.awt.Container.processEvent(Container.java:2236)
	at java.awt.Component.dispatchEventImpl(Component.java:4889)
	at java.awt.Container.dispatchEventImpl(Container.java:2294)
	at java.awt.Component.dispatchEvent(Component.java:4711)
	at java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4888)
	at java.awt.LightweightDispatcher.processMouseEvent(Container.java:4525)
	at java.awt.LightweightDispatcher.dispatchEvent(Container.java:4466)
	at java.awt.Container.dispatchEventImpl(Container.java:2280)
	at java.awt.Window.dispatchEventImpl(Window.java:2746)
	at java.awt.Component.dispatchEvent(Component.java:4711)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:758)
	at java.awt.EventQueue.access$500(EventQueue.java:97)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.awt.EventQueue$3.run(EventQueue.java:703)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:80)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:90)
	at java.awt.EventQueue$4.run(EventQueue.java:731)
	at java.awt.EventQueue$4.run(EventQueue.java:729)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:80)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:728)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)
Caused by: net.sf.launch4j.ExecException: Exec failed (2): /home/pi/Desktop/processing-3.3.5/modes/java/application/launch4j/bin/windres --preprocessor=cat -J rc -O coff -F pe-i386 /tmp/launch4j2560377606394614871rc /tmp/launch4j6100233142982000968o
	at net.sf.launch4j.Util.exec(Util.java:156)
	at net.sf.launch4j.Cmd.exec(Builder.java:215)
	at net.sf.launch4j.Builder.build(Builder.java:100)
	... 57 more

Total time: 1 second
```

## Steps to Reproduce

1. Create/load up any sketch
2. Go to File > Export Application...
3. Select Windows
4. Click Export

## Your Environment

 

* Processing version: 3.3.6
* Operating System and OS version: Raspbian Jessie July 2017 Release
* Other information: OS Running on a Raspberry Pi 3

## Possible Causes / Solutions

If you look at the first lines of the log, you can see this:

```
102 RCDATA BEGIN "This application was configured to use a bundled Java Runtime Environment but the runtime is missing or corrupted.\0" END
```
I don't know if it can be fixed then, because Raspbian has OpenJDK, not Java by itself. It might be fixed by changing some names in the armv6hf build.

**Comment by benfry** [comment]
@gohai we'll need a version of launch4j with an `ld` and `windres` that works on ARM.

**Comment by gohai** [comment]
Good catch. I'll grab binutils-2.19 and compile on ARM with `--target=i686-pc-mingw32`, as I believe this is how the existing variants were built.

**Comment by uwemock** [comment]
Any news on how to get Launch4J working on a Pi?
I'm trying to run it from the maven plugin.

**Comment by gohai** [comment]
@liquid600pgm What's not working with Launch4J? I am running exported application all the time using the Processing IDE on the Pi.

**Comment by gohai** [comment]
Fixed in PR #5488.
